src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -326,15 +326,23 @@ trait Infer extends Checkable {
         (samFun ne NoType) && isCompatible(tp, samFun)
       }
 
-      val tp1 = methodToExpressionTp(tp)
+      // can only compare if both types are repeated or neither is (T* is not actually a first-class type, even though it has a BTS and thus participates in subtyping)
+      (!isRepeatedParamType(tp) || isRepeatedParamType(pt)) && {
+        val tp1 = methodToExpressionTp(tp)
+
+        ((tp1 weak_<:< pt)
+         || isCoercible(tp1, pt)
+         || isCompatibleByName(tp, pt)
+         || isCompatibleSam(tp, pt)
+        )
+      }
+    }
 
-      (    (tp1 weak_<:< pt)
-        || isCoercible(tp1, pt)
-        || isCompatibleByName(tp, pt)
-        || isCompatibleSam(tp, pt)
-      )
+    def isCompatibleArgs(tps: List[Type], pts: List[Type]) = {
+      val res = (tps corresponds pts)(isCompatible)
+//      println(s"isCompatibleArgs $res : $tps <:< $pts")
+      res
```

Comment:
I think you need an `@inline printing[A](msg, a: => A): A` thingie with elidable printing.

Reply:
How I learned to stop worrying about performance and have debug output when I need it. Comments! Higher-tech would be nice, but it would also look like regular code, when it isn't really.